src/lsm/level_data_iterator.zig

```diff
@@ -124,12 +124,12 @@ pub fn LevelTableValueBlockIteratorType(comptime Table: type, comptime Storage:
                 // Refill `table_data_iterator` before calling `table_next`.
                 const table_ref = it.context.tables[it.table_index];
                 it.callback = .{ .level_next = callback };
-                it.context.grid.read_block_from_cache_or_storage(
-                    on_level_next,
+                it.context.grid.read_block(
+                    .{ .from_local_or_global_storage = on_level_next },
                     &it.read,
                     table_ref.table_info.address,
                     table_ref.table_info.checksum,
-                    .index,
+                    .{ .cache_read = true, .cache_write = true },
```

Comment:
:thinking: it seems lsm pretty-much always wants a coherent reaad. 

Maybe its worth to add a helper function for this, which just delegates to `read_block` with appropriate arguments? 

I think from the perspective of LSM we want to work with a simple grid, which just has blocks. We don't want to thing about local_or_global storage and cache coherency. 

Basiaclly, we only need advanced read functionality for scrubber & replica's repiar, lsm could use simpler API. 


But also having `read_block` and `read_block_advanced` sound kinda stupid, so maybe we don't need it? But also, grepping for `read_block_advanced` is easier than grepping for `read_block` and noticing odd arguments...

Reply:
Yes, LSM always wants a coherent read.

I went back and forth on this with the exact same reasons you gave :laughing: 
I ended up on just `read_block()` since it keeps things uniform, but I don't feel too strongly about it.

Reply:
Yeah, let's keep as is then. But just one more option in case its a clear tiebreaker for you (it isn't for me):


We could go very specific here, make `read` private, and add

```
pub fn read_lsm
pub fn read_scrubber
pub fn read_replica
```

with appropriate specialization. That way all the callers of grid are sort-of inside the grid, and the logic of what's to cache or not cache stays encapsulated. 

Though, don't like it much --- I bet at some point something somewhere would want to use an odd combination of read flags.

Reply:
Yeah, in particular I could see us experimenting with different cache options (and even adding more granular ones) for compaction in the future. For example, don't cache compaction input tables since they won't be needed for long.